fix: Improve button style consistency

dcalhoun opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

Relates to https://github.com/Automattic/studio/pull/214#discussion_r1635129557.

## Proposed Changes

Introduce a new `outlined` button variant.

Overloading the in-use `tertiary` button variant styles resulted in
unexpected and inconsistent styles in the UI --- e.g., the "Add site"
dialog. Introducing a new `outlined` variant achieves the original goal
of a new button style that is now used in both the sidebar and AI
assistant code blocks.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**Expected outline button styles display**

1. Verify the sidebar's "Add site" button renders correct styles.
1. Navigate to the "Assistant" tab.
1. Send a "Show me some code" message.
1. Verify the "Copy" and "Run"[^1] buttons render correct styles.

| Before | After |
| - | - |
| ![before-inline-code](https://github.com/Automattic/studio/assets/438664/0d3217ed-b855-4b6e-afc9-06b399525420) | ![after-inline-code](https://github.com/Automattic/studio/assets/438664/f1a11032-9e9c-4146-83bf-4e56c3b4f97c) |

[^1]: The "Run" button is currently commented in the source code and thus hidden. 

**Expected tertiary button styles display**

1. Click "Add site."
1. Verify the "Cancel" button renders correct styles.

| Before | After |
| - | - |
| ![before-add-site](https://github.com/Automattic/studio/assets/438664/0693e9bd-7b21-4f3b-a908-ba5cead820a9) | ![after-add-site](https://github.com/Automattic/studio/assets/438664/0d11a792-ee46-49db-95be-d14581f0be16) |

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

dcalhoun committed:
fix: Improve button style consistency

Overloading the in-use `tertiary` button variant styles resulted in
unexpected and inconsistent styles in the UI --- e.g., the "Add site"
dialog. Introducing a new `outlined` variant achieves the original goal
of a new button style that is now used in both the sidebar and AI
assistant code blocks.

dcalhoun commented:
👋🏻 @matt-west. The changes I propose do _not_ include the reduced button padding captured in the [design proposal](https://github.com/Automattic/studio/pull/214/files#r1636051435) as I was uncertain if we intended to apply that to the "Add site" button as well. 

My thought is that the current padding used for all existing buttons looks fine for the inline code buttons (see images in the PR description). If you disagree, we could introduce additional one-off button style modifications for the padding of the inline code buttons. WDYT?

matt-west commented:
Thanks for implementing this change @dcalhoun.

I think the code buttons need the reduced padding, they're a bit too dominating at the moment. We should keep the existing padding for the `Add site` button though. Applying one-off modifications to the code buttons makes sense to me. 👍

dcalhoun committed:
feat: Reduce inline code button size

The default button size is too large and dominating for the context of
inline code within AI assistant responses.

fluiddot reviewed (approved):
LGTM 🎊 !
